File: Longer_Contiguous_Segments_of_Ones_than_Zeros.cpp
class Solution {
public:
    bool checkZeroOnes(string s) {
        int oneMax = 0;
        int zeroMax = 0;
        int oneStreak = 0;
        int zeroStreak = 0;

        for(char n : s){
            if(n == '1'){
                zeroStreak = 0;
                oneStreak++;
                oneMax = max(oneStreak, oneMax);
            }
            else{
                oneStreak = 0;
                zeroStreak++;
                zeroMax = max(zeroStreak, zeroMax);
            }
        }

       return oneMax > zeroMax; 
    }
};

File: Longer_Contiguous_Segments_of_Ones_than_Zeros.java
class Solution {
    public boolean checkZeroOnes(String s) {
        int ones = 0, zeroes = 0, consecOnes = 0, consecZeroes = 0;
        for (char c: s.toCharArray()) {
            if (c == '1') {
                consecOnes++;
                consecZeroes = 0;
                ones = Math.max(ones, consecOnes);
            } else {
                consecZeroes++;
                consecOnes = 0;
                zeroes = Math.max(zeroes, consecZeroes);
            }
        }
        return ones > zeroes;
    }
}
